analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.Arithmetic.cs

```diff
@@ -141,6 +143,48 @@ private static NumberConstraint CalculateMultiply(NumberConstraint left, NumberC
         }
     }
 
+    private static BigInteger? CalculateXorMin(NumberConstraint left, NumberConstraint right)
+    {
+        if (left.IsPositive && right.IsPositive)
+        {
+            return left.CanOverlap(right) ? 0 : Min(left.Min - right.Max, right.Min - left.Max);
+        }
+        else if (left.IsNegative && right.IsNegative)
+        {
+            return left.CanOverlap(right) ? 0 : Min(left.Max - right.Min, right.Max - left.Min);
+        }
+        return null;
```

Comment:
This should also be true, I think:
```suggestion
        if (left.Min.HasValue && right.Min.HasValue)
        {
            NegativeMagnitude(BigInteger.Min(left.Min.Value, right.Min.Value));
        }
        return null;
```

Reply:
Try this, it actually fixes two of the existing UTs.
```suggestion
        else if ((left.IsPositive || right.IsNegative) && left.Max.HasValue && right.Min.HasValue)
        {
            return NegativeMagnitude(-BigInteger.Max(left.Max.Value, BigInteger.Abs(right.Min.Value)));
        }
        else if ((left.IsNegative || right.IsPositive) && left.Min.HasValue && right.Max.HasValue)
        {
            return NegativeMagnitude(-BigInteger.Max(BigInteger.Abs(left.Min.Value), right.Max.Value));
        }
        else if (left.Min.HasValue && left.Max.HasValue && right.Min.HasValue && right.Max.HasValue)
        {
            return NegativeMagnitude(-BigInteger.Max(
                BigInteger.Max(BigInteger.Abs(left.Min.Value), BigInteger.Abs(right.Max.Value)),
                BigInteger.Max(BigInteger.Abs(left.Max.Value), BigInteger.Abs(right.Min.Value))));
        }
        else
        {
            return null;
        }
```

Reply:
Tried and worked for existing use cases. However, after some analysis, I could find a case that breaks your code:
```cs
[DataRow("i >= -3 && i <= -1 && j >= -4 && j <=-2", 0, 3)]
// Expected SETestContext.CreateCS(code, "int i, int j").Validator.TagValue("Value") 
// to have constraints {NotNull, Number from 0 to 3}, but {Number from 0 to 3} are missing 
// and additional {Number from 3 to 7} are present. Actual are {NotNull, Number from 3 to 7}.
```

Even if I got `CalculateXorMax` (see comments below), an issue surely also affects `CalculateXorMin`.
So, I am not applying these changes, and only keep basic cases where we have solid understanding of how to calculate lower bounds.

Reply:
The test case is not affected by the code I suggested. When I add my code suggestion to the current implementation it does what I expect: It improves two test cases ("i ==  4 && j >= -6" and "i == -4 && j <=  6").